CaCertsTest.test_request_with_verifypeer fails

## What did you try to do?
While building the official packages for openSUSE/Tumbleweed, I run the test suite. The test CaCertsTest.test_request_with_verifypeer failed with this error:

```
[   28s] =================================== FAILURES ===================================
[   28s] ___________________ CaCertsTest.test_request_with_verifypeer ___________________
[   28s] 
[   28s] self = <tests.cadata_test.CaCertsTest testMethod=test_request_with_verifypeer>
[   28s] 
[   28s]     @util.only_ssl_backends('openssl')
[   28s]     def test_request_with_verifypeer(self):
[   28s]         with open(os.path.join(os.path.dirname(__file__), 'certs', 'ca.crt'), 'rb') as stream:
[   28s]             cadata = stream.read().decode('ASCII')
[   28s]         self.curl.setopt(pycurl.URL, 'https://localhost:8384/success')
[   28s]         sio = util.BytesIO()
[   28s]         self.curl.set_ca_certs(cadata)
[   28s]         self.curl.setopt(pycurl.WRITEFUNCTION, sio.write)
[   28s]         # self signed certificate, but ca cert should be loaded
[   28s]         self.curl.setopt(pycurl.SSL_VERIFYPEER, 1)
[   28s] >       self.curl.perform()
[   28s] E       pycurl.error: (60, 'SSL certificate problem: self-signed certificate')
[   28s] 
[   28s] tests/cadata_test.py:31: error
[   28s] =============================== warnings summary ===============================
[   28s] tests/cadata_test.py::CaCertsTest::test_request_with_verifypeer
[   28s]   /usr/lib/python3.9/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-1
[   28s]   
[   28s]   Traceback (most recent call last):
[   28s]     File "/usr/lib64/python3.9/threading.py", line 980, in _bootstrap_inner
[   28s]       self.run()
[   28s]     File "/home/abuild/rpmbuild/BUILD/pycurl-7.45.2/tests/runwsgi.py", line 69, in run
[   28s]       bottle.run(self.app, server=self.server, quiet=True)
[   28s]     File "/usr/bin/bottle.py", line 3175, in run
[   28s]       server.run(app)
[   28s]     File "/home/abuild/rpmbuild/BUILD/pycurl-7.45.2/tests/runwsgi.py", line 34, in run
[   28s]       self.srv = self.make_server(handler)
[   28s]     File "/home/abuild/rpmbuild/BUILD/pycurl-7.45.2/tests/runwsgi.py", line 24, in make_server
[   28s]       srv = make_server(self.host, self.port, handler, **self.options)
[   28s]     File "/usr/lib64/python3.9/wsgiref/simple_server.py", line 154, in make_server
[   28s]       server = server_class((host, port), handler_class)
[   28s]     File "/usr/lib64/python3.9/socketserver.py", line 452, in __init__
[   28s]       self.server_bind()
[   28s]     File "/usr/lib64/python3.9/wsgiref/simple_server.py", line 50, in server_bind
[   28s]       HTTPServer.server_bind(self)
[   28s]     File "/usr/lib64/python3.9/http/server.py", line 137, in server_bind
[   28s]       socketserver.TCPServer.server_bind(self)
[   28s]     File "/usr/lib64/python3.9/socketserver.py", line 466, in server_bind
[   28s]       self.socket.bind(self.server_address)
[   28s]   OSError: [Errno 98] Address already in use
[   28s]   
[   28s]     warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
[   28s] 
[   28s] -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[   28s] ===Flaky Test Report===
[   28s] 
[   28s] test_pause_via_call passed 1 out of the required 1 times. Success!
[   28s] test_pause_via_return passed 1 out of the required 1 times. Success!
[   28s] test_post_byte_buffer passed 1 out of the required 1 times. Success!
[   28s] test_post_fields_with_ampersand passed 1 out of the required 1 times. Success!
[   28s] test_post_file passed 1 out of the required 1 times. Success!
[   28s] test_post_list_of_list_of_lists passed 1 out of the required 1 times. Success!
[   28s] test_post_list_of_tuple_of_tuples passed 1 out of the required 1 times. Success!
[   28s] test_post_multiple_fields passed 1 out of the required 1 times. Success!
[   28s] test_post_single_field passed 1 out of the required 1 times. Success!
[   28s] test_post_tuple_of_lists_of_tuples passed 1 out of the required 1 times. Success!
[   28s] test_post_tuple_of_tuple_of_lists passed 1 out of the required 1 times. Success!
[   28s] test_post_tuple_of_tuples_of_tuples passed 1 out of the required 1 times. Success!
[   28s] test_post_unicode_buffer passed 1 out of the required 1 times. Success!
[   28s] test_post_with_null_byte passed 1 out of the required 1 times. Success!
[   28s] 
[   28s] ===End Flaky Test Report===
[   28s] =========================== short test summary info ============================
[   28s] FAILED tests/cadata_test.py::CaCertsTest::test_request_with_verifypeer - pycu...
[   28s] ===== 1 failed, 374 passed, 7 skipped, 44 deselected, 1 warning in 13.57s ======
```

## What happened?
The test failed.

See also [complete build log](https://github.com/pycurl/pycurl/files/12885820/_log.txt) with all packages used and steps taken to reproduce.

## What is the PycURL version?

7.45.2


## What is your Python version?

various (3.9.18, 3.10.13, 3.11.5)


## What is your operating system and its version?
openSUSE/Tumbleweed (as of today, see the above log for the complete list of versions of packages used).

Yes (with `pycurl-7.45.3`):

```
[   37s] ___________________ CaCertsTest.test_request_with_verifypeer ___________________
[   37s] 
[   37s] self = <tests.cadata_test.CaCertsTest testMethod=test_request_with_verifypeer>
[   37s] 
[   37s]     @util.only_ssl_backends('openssl')
[   37s]     def test_request_with_verifypeer(self):
[   37s]         with open(os.path.join(os.path.dirname(__file__), 'certs', 'ca.crt'), 'rb') as stream:
[   37s]             cadata = stream.read().decode('ASCII')
[   37s]         self.curl.setopt(pycurl.URL, 'https://localhost:8384/success')
[   37s]         sio = util.BytesIO()
[   37s]         self.curl.set_ca_certs(cadata)
[   37s]         self.curl.setopt(pycurl.WRITEFUNCTION, sio.write)
[   37s]         # self signed certificate, but ca cert should be loaded
[   37s]         self.curl.setopt(pycurl.SSL_VERIFYPEER, 1)
[   37s] >       self.curl.perform()
[   37s] E       pycurl.error: (60, 'SSL peer certificate or SSH remote key was not OK')
[   37s] 
[   37s] tests/cadata_test.py:31: error
[   37s] =============================== warnings summary ===============================
[   37s] tests/cadata_test.py::CaCertsTest::test_request_with_verifypeer
[   37s]   /usr/lib/python3.10/site-packages/_pytest/threadexception.py:82: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-1
[   37s]   
[   37s]   Traceback (most recent call last):
[   37s]     File "/usr/lib64/python3.10/threading.py", line 1016, in _bootstrap_inner
[   37s]       self.run()
[   37s]     File "/home/abuild/rpmbuild/BUILD/pycurl-7.45.3/tests/runwsgi.py", line 75, in run
[   37s]       self.server.run(self.app)
[   37s]     File "/home/abuild/rpmbuild/BUILD/pycurl-7.45.3/tests/runwsgi.py", line 40, in run
[   37s]       self.srv = self.make_server(handler)
[   37s]     File "/home/abuild/rpmbuild/BUILD/pycurl-7.45.3/tests/runwsgi.py", line 30, in make_server
[   37s]       srv = make_server(self.host, self.port, handler, **self.options)
[   37s]     File "/usr/lib64/python3.10/wsgiref/simple_server.py", line 154, in make_server
[   37s]       server = server_class((host, port), handler_class)
[   37s]     File "/usr/lib64/python3.10/socketserver.py", line 452, in __init__
[   37s]       self.server_bind()
[   37s]     File "/usr/lib64/python3.10/wsgiref/simple_server.py", line 50, in server_bind
[   37s]       HTTPServer.server_bind(self)
[   37s]     File "/usr/lib64/python3.10/http/server.py", line 137, in server_bind
[   37s]       socketserver.TCPServer.server_bind(self)
[   37s]     File "/usr/lib64/python3.10/socketserver.py", line 466, in server_bind
[   37s]       self.socket.bind(self.server_address)
[   37s]   OSError: [Errno 98] Address already in use
[   37s]   
[   37s]     warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
[   37s] 
[   37s] -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[   37s] ===Flaky Test Report===
[   37s] 
[   37s] test_pause_via_call passed 1 out of the required 1 times. Success!
[   37s] test_pause_via_return passed 1 out of the required 1 times. Success!
[   37s] test_post_byte_buffer passed 1 out of the required 1 times. Success!
[   37s] test_post_fields_with_ampersand passed 1 out of the required 1 times. Success!
[   37s] test_post_file passed 1 out of the required 1 times. Success!
[   37s] test_post_list_of_list_of_lists passed 1 out of the required 1 times. Success!
[   37s] test_post_list_of_tuple_of_tuples passed 1 out of the required 1 times. Success!
[   37s] test_post_multiple_fields passed 1 out of the required 1 times. Success!
[   37s] test_post_single_field passed 1 out of the required 1 times. Success!
[   37s] test_post_tuple_of_lists_of_tuples passed 1 out of the required 1 times. Success!
[   37s] test_post_tuple_of_tuple_of_lists passed 1 out of the required 1 times. Success!
[   37s] test_post_tuple_of_tuples_of_tuples passed 1 out of the required 1 times. Success!
[   37s] test_post_unicode_buffer passed 1 out of the required 1 times. Success!
[   37s] test_post_with_null_byte passed 1 out of the required 1 times. Success!
[   37s] 
[   37s] ===End Flaky Test Report===
[   37s] =========================== short test summary info ============================
[   37s] FAILED tests/cadata_test.py::CaCertsTest::test_request_with_verifypeer - pycu...
[   37s] = 1 failed, 384 passed, 8 skipped, 60 deselected, 1 xpassed, 1 warning in 15.42s =
```

[Complete build log](https://github.com/user-attachments/files/18050053/_log.txt)

This is master at 891cd7d:

```
[   39s] =================================== FAILURES ===================================
[   39s] ___________________ CaCertsTest.test_request_with_verifypeer ___________________
[   39s] 
[   39s] self = <tests.cadata_test.CaCertsTest testMethod=test_request_with_verifypeer>
[   39s] 
[   39s]     @util.only_ssl_backends('openssl')
[   39s]     def test_request_with_verifypeer(self):
[   39s]         with open(os.path.join(os.path.dirname(__file__), 'certs', 'ca.crt'), 'rb') as stream:
[   39s]             cadata = stream.read().decode('ASCII')
[   39s]         self.curl.setopt(pycurl.URL, 'https://localhost:8384/success')
[   39s]         sio = util.BytesIO()
[   39s]         self.curl.set_ca_certs(cadata)
[   39s]         self.curl.setopt(pycurl.WRITEFUNCTION, sio.write)
[   39s]         # self signed certificate, but ca cert should be loaded
[   39s]         self.curl.setopt(pycurl.SSL_VERIFYPEER, 1)
[   39s] >       self.curl.perform()
[   39s] E       pycurl.error: (60, 'SSL peer certificate or SSH remote key was not OK')
[   39s] 
[   39s] tests/cadata_test.py:31: error
[   39s] =============================== warnings summary ===============================
[   39s] tests/cadata_test.py::CaCertsTest::test_request_with_verifypeer
[   39s]   /usr/lib/python3.10/site-packages/_pytest/threadexception.py:82: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-1
[   39s]   
[   39s]   Traceback (most recent call last):
[   39s]     File "/usr/lib64/python3.10/threading.py", line 1016, in _bootstrap_inner
[   39s]       self.run()
[   39s]     File "/home/abuild/rpmbuild/BUILD/pycurl-7.45.3+git.1733623296.891cd7d/tests/runwsgi.py", line 75, in run
[   39s]       self.server.run(self.app)
[   39s]     File "/home/abuild/rpmbuild/BUILD/pycurl-7.45.3+git.1733623296.891cd7d/tests/runwsgi.py", line 40, in run
[   39s]       self.srv = self.make_server(handler)
[   39s]     File "/home/abuild/rpmbuild/BUILD/pycurl-7.45.3+git.1733623296.891cd7d/tests/runwsgi.py", line 30, in make_server
[   39s]       srv = make_server(self.host, self.port, handler, **self.options)
[   39s]     File "/usr/lib64/python3.10/wsgiref/simple_server.py", line 154, in make_server
[   39s]       server = server_class((host, port), handler_class)
[   39s]     File "/usr/lib64/python3.10/socketserver.py", line 452, in __init__
[   39s]       self.server_bind()
[   39s]     File "/usr/lib64/python3.10/wsgiref/simple_server.py", line 50, in server_bind
[   39s]       HTTPServer.server_bind(self)
[   39s]     File "/usr/lib64/python3.10/http/server.py", line 137, in server_bind
[   39s]       socketserver.TCPServer.server_bind(self)
[   39s]     File "/usr/lib64/python3.10/socketserver.py", line 466, in server_bind
[   39s]       self.socket.bind(self.server_address)
[   39s]   OSError: [Errno 98] Address already in use
[   39s]   
[   39s]     warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
[   39s] 
[   39s] -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[   39s] ===Flaky Test Report===
[   39s] 
[   39s] test_pause_via_call passed 1 out of the required 1 times. Success!
[   39s] test_pause_via_return passed 1 out of the required 1 times. Success!
[   39s] test_post_byte_buffer passed 1 out of the required 1 times. Success!
[   39s] test_post_fields_with_ampersand passed 1 out of the required 1 times. Success!
[   39s] test_post_file passed 1 out of the required 1 times. Success!
[   39s] test_post_list_of_list_of_lists passed 1 out of the required 1 times. Success!
[   39s] test_post_list_of_tuple_of_tuples passed 1 out of the required 1 times. Success!
[   39s] test_post_multiple_fields passed 1 out of the required 1 times. Success!
[   39s] test_post_single_field passed 1 out of the required 1 times. Success!
[   39s] test_post_tuple_of_lists_of_tuples passed 1 out of the required 1 times. Success!
[   39s] test_post_tuple_of_tuple_of_lists passed 1 out of the required 1 times. Success!
[   39s] test_post_tuple_of_tuples_of_tuples passed 1 out of the required 1 times. Success!
[   39s] test_post_unicode_buffer passed 1 out of the required 1 times. Success!
[   39s] test_post_with_null_byte passed 1 out of the required 1 times. Success!
[   39s] 
[   39s] ===End Flaky Test Report===
[   39s] =========================== short test summary info ============================
[   39s] FAILED tests/cadata_test.py::CaCertsTest::test_request_with_verifypeer - pycu...
[   39s] ===== 1 failed, 386 passed, 8 skipped, 62 deselected, 1 warning in 15.28s ======
```
[_log.txt](https://github.com/user-attachments/files/18059283/_log.txt)

--- a/tests/close_socket_cb_test.py
+++ b/tests/close_socket_cb_test.py
@@ -96,6 +96,7 @@ def test_closesocketfunction_on_close(app):
 
     curl = util.DefaultCurl()
     curl.setopt(curl.URL, f"{app}/success")
+    curl.setopt(pycurl.IPRESOLVE, pycurl.IPRESOLVE_V4)
     curl.setopt(pycurl.FORBID_REUSE, False)
     curl.setopt(pycurl.CONNECT_ONLY, True)
     curl.setopt(pycurl.CLOSESOCKETFUNCTION, closesocketfunction)
@@ -120,6 +121,7 @@ def test_closesocketfunction_on_dealloc(app):
 
     curl = util.DefaultCurl()
     curl.setopt(curl.URL, f"{app}/success")
+    curl.setopt(pycurl.IPRESOLVE, pycurl.IPRESOLVE_V4)
     curl.setopt(pycurl.FORBID_REUSE, False)
     curl.setopt(pycurl.CONNECT_ONLY, True)
     curl.setopt(pycurl.CLOSESOCKETFUNCTION, closesocketfunction)

--- a/tests/open_socket_cb_test.py
+++ b/tests/open_socket_cb_test.py
@@ -2,6 +2,7 @@ from __future__ import annotations
 
 import socket
 from io import BytesIO
+from urllib.parse import urlparse
 
 import pycurl
 import pytest
@@ -45,7 +46,8 @@ def test_socket_open_ipv6(curl, app):
         return s
 
     curl.setopt(pycurl.OPENSOCKETFUNCTION, on_open)
-    curl.setopt(curl.URL, f"{app.replace('127.0.0.1', '[::1]')}/success")
+    port = urlparse(app).port
+    curl.setopt(curl.URL, f"http://[::1]:{port}/success")
     sio = BytesIO()
     curl.setopt(pycurl.WRITEFUNCTION, sio.write)
     with pytest.raises(pycurl.error):

--- a/tests/test_callback_signals.py
+++ b/tests/test_callback_signals.py
@@ -61,6 +61,7 @@ def test_opensocket_callback_keyboard_interrupt(callback_curl):
         called["called"] = True
         raise KeyboardInterrupt()
 
+    callback_curl.setopt(pycurl.IPRESOLVE, pycurl.IPRESOLVE_V4)
     callback_curl.setopt(pycurl.OPENSOCKETFUNCTION, opensocket_function)
 
     with pytest.raises(KeyboardInterrupt):

